okhttp/src/main/java/okhttp3/Cache.java

```diff
@@ -733,41 +733,20 @@ private static int readInt(BufferedSource source) throws IOException {
     }
   }
 
-  private static class CacheResponseBody extends ResponseBody {
+  private static class CacheResponseBody extends Body {
     private final DiskLruCache.Snapshot snapshot;
-    private final BufferedSource bodySource;
-    private final String contentType;
-    private final String contentLength;
 
-    public CacheResponseBody(final DiskLruCache.Snapshot snapshot,
-        String contentType, String contentLength) {
+    public CacheResponseBody(final DiskLruCache.Snapshot snapshot, String contentType,
+        String contentLength) {
+      super(contentType != null ? MediaType.parse(contentType) : null,
```

Comment:
I nolove this. I think the other form with abstract methods was easier to reason about.